scripts/shared/lib/deploy_bundle

```diff
@@ -0,0 +1,249 @@
+spec:
+  targetNamespaces:
+    - ${SUBM_NS}
+EOF
+
+  # Debug
+  kubectl get og -n "${SUBM_NS}" --ignore-not-found
+
+  echo "[INFO](${cluster}) Catalog source created"
+}
+
+function install_bundle() {
+  local installPlan
+
+  # Delete previous catalogSource and Subscription
+  kubectl delete sub/submariner-subscription -n "${SUBM_NS}" --wait --ignore-not-found
+
+  # Create the Subscription (Approval should be Manual not Automatic in order to pin the bundle version)
+  echo "[INFO](${cluster}) Install the bundle..."
+cat <<EOF | kubectl apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: submariner-subscription
+  namespace: ${SUBM_NS}
+spec:
+  channel: ${CHANNEL}
+  installPlanApproval: Manual
+  name: ${OPERATOR_NAME}
+  source: submariner-catalog-source
+  sourceNamespace: ${MARKETPLACE_NAMESPACE}
+  startingCSV: ${OPERATOR_NAME}.v${VERSION}
+EOF
+
+  # Manual Approve
+  echo "[INFO](${cluster}) Approve the installPlan..."
+  kubectl wait --for condition=InstallPlanPending --timeout=5m -n "${SUBM_NS}" subs/submariner-subscription || (echo "[ERROR](${cluster}) InstallPlan not found."; exit 1)
+  installPlan=$(kubectl get subscriptions.operators.coreos.com submariner-subscription -n "${SUBM_NS}" -o jsonpath='{.status.installPlanRef.name}')
+  if [ -n "${installPlan}" ]; then
+    kubectl patch installplan -n "${SUBM_NS}" "${installPlan}" -p '{"spec":{"approved":true}}' --type merge
+  fi
+
+  # Debug
+  kubectl get sub -n "${SUBM_NS}" --ignore-not-found
+  kubectl get installplan -n "${SUBM_NS}" --ignore-not-found
+  kubectl get csv -n "${SUBM_NS}" --ignore-not-found
+  kubectl get pods -n "${SUBM_NS}" --ignore-not-found
+
+  echo "[INFO](${cluster}) Bundle installed"
+}
+
+function setup_broker() {
+  local brokerClientSecret
+
+  if ! (timeout 5m bash -c "until kubectl --context=${cluster} get crds brokers.submariner.io > /dev/null 2>&1; do sleep 10; done"); then
+      echo "[ERROR] Broker CRD was not found."
+      exit 1
+  fi
+
+  # Create the broker Namespace & RBAC
+  echo "[INFO] Deploy the broker"
+  [ -d /tmp/submariner-operator ] && rm -Rf /tmp/submariner-operator
+  git clone --depth 1 --single-branch --branch "${BASE_BRANCH}" https://github.com/submariner-io/submariner-operator /tmp/submariner-operator
+  kubectl apply -k /tmp/submariner-operator/config/broker -n ${BROKER_NAMESPACE}
+
+  # Enable the service-discovery component if defined
+  local components="connectivity"
+  components+=$([[ ${service_discovery} == "true" ]] && echo ", service-discovery" || echo "")
+  components+=$([[ ${globalnet} == "true" ]] && echo ", globalnet" || echo "")
+
+  ### Create the Broker instance
+cat <<EOF | kubectl apply -f -
+apiVersion: submariner.io/v1alpha1
+kind: Broker
+metadata:
+  name: submariner-broker
+  namespace: ${SUBM_NS}
+spec:
+  components: [${components}]
+  defaultGlobalnetClusterSize: 8192
+  globalnetEnabled: ${globalnet}
+  globalnetCIDRRange: 169.254.0.0/16
+EOF
```

Comment:
This would be better if it was in a real YAML file which gets validated. You can use `render_template` function from `clusters.sh`